students/elmar_m/lesson06/test_mailroom.py

```diff
@@ -0,0 +1,79 @@
+#!/usr/bin/env python3
+
+'''
+file: test_mailroom.py
+elmar_m / 22e88@mailbox.org
+Lesson06: Mailroom Exercise Part 4/ unittesting
+'''
+
+import mailroom as m
+from unittest import TestCase
+from unittest import main as utmain
+from unittest.mock import patch
+from unittest.mock import MagicMock
+from io import StringIO
+import os
+import sys
+
+def test_writefile():
+    content = 'This is a test\n'
+    assert m.writefile('test', content) is True
+
+
+def test_mail():
+    assert m.mail() is True
+
+
+def test_list_donors():
+    assert m.list_donors() is True
+
+
+class mytestclass(TestCase):
+    def test_add(self):
+        '''
+        Test a function which expects STDIN input from users.
+        Simulate several inputs by manipulating the builtin function input    
+        with unittest.mock.patch(). Use the side_effect parameter to
+        pass in the fake input you want to test.
+        side_effect expects a list as it's argument.
+        '''
+        with patch('builtins.input', side_effect=['berta']) as fake_input:
+            with patch('sys.stdout', new=StringIO()) as fake_output:
+                m.add_amount = MagicMock(return_value=True)
+                m.add()                
+                self.assertEqual(fake_output.getvalue(), '>> berta already in list\n')
+
+        with patch('builtins.input', side_effect=['nobody']) as fake_input:
+            with patch('sys.stdout', new=StringIO()) as fake_output:
+                m.add_amount = MagicMock(return_value=True)
+                m.add()
+                self.assertEqual(fake_output.getvalue(), '>> nobody not in list, adding it\n')
```

Comment:
I don't think you need to mock print statements or `add_amount`

you should be able to verify that a donor was added to `donors` or a donation amount was added to existing donor in `donors`